substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrNativeEventWriter.java

```diff
@@ -80,11 +80,11 @@ public static UnsignedWord endEventWrite(JfrNativeEventWriterData data, boolean
         UnsignedWord written = getUncommittedSize(data);
         if (large) {
             // Write a 4 byte size and commit the event if any payload was written.
-            if (written.aboveThan(Integer.BYTES)) {
+            if (written.belowOrEqual(MAX_PADDED_INT_VALUE)) {
```

Comment:
The logic in this method is in general not particularly consistent at the moment. I think it should be changed to the following:
- The event should only be committed if any event payload (i.e., anything besides the event size that is reserved early on) was written.
- execute `reset(data)` and return 0 if the event was not committed (e.g., because the event was too large)